packages/wonder-blocks-button/components/button.md

```diff
@@ -93,171 +141,280 @@ const styles = StyleSheet.create({
+        size="small"
+    >
+        Label
+    </Button>
+    <Button
+        style={styles.sideMargins}
+        onClick={(e) => console.log("Hello, world!")}
+        kind="secondary"
+        size="small"
+    >
+        Label
+    </Button>
+    <Button
+        style={styles.sideMargins}
+        onClick={(e) => console.log("Hello, world!")}
+        kind="tertiary"
+        size="small"
+    >
+        Label
+    </Button>
+</View>
+```
+
+Buttons can have an `href` or an `onClick` handler or both.
+
+Being able to use both is necessary to support marking conversions in A/B tests.
+```jsx
+const {View} = require("@khanacademy/wonder-blocks-core");
+const {StyleSheet} = require("aphrodite");
+
+const styles = StyleSheet.create({
+    row: {
+        flexDirection: "row",
     },
     sideMargins: {
         marginRight: 10,
+    }
+});
+
+<View style={styles.row}>
+    <Button
+        href="#button-1"
+        style={styles.sideMargins}
+    >
+        href
+    </Button>
+    <Button
+        onClick={(e) => console.log("Hello, world!")}
+        style={styles.sideMargins}
+    >
+        onClick
+    </Button>
+    <Button
+        href="#button-1"
+        onClick={(e) => console.log("Hello, world!")}
+        style={styles.sideMargins}
+    >
+        both
+    </Button>
+</View>
+```
+
+Buttons can have a `style` props which supports width, position, margin,
+and flex styles:
+
+TODO(kevinb): link to front matter section describing the `style` prop in
+more detail.
+
+### Best Practices
+
+In vertical layouts, buttons will stretch horizontally to fill the available
+space.  This is probably not what you want unless you're on a very narrow
+screen.
+
+TODO(kevinb): add section on media queries.
+```jsx
+const {View} = require("@khanacademy/wonder-blocks-core");
+
+<View>
+    <Button>
+        Label
+    </Button>
+</View>
+```
+
+This can be corrected by applying appropriate flex styles to the container.
+
+TODO(kevinb): link to front matter section describing why we use flexbox for
+everything.
+```jsx
+const {View} = require("@khanacademy/wonder-blocks-core");
+const {StyleSheet} = require("aphrodite");
+
+const styles = StyleSheet.create({
+    column: {
+        alignItems: "flex-start",
+    },
+    row: {
+        flexDirection: "row",
+    },
+    gap: {
+        height: 16,
+    },
+    button: {
+        marginRight: 10,
     },
 });
 
+<View>
+    <View style={styles.row}>
+        <Button>
+            Button in a row
+        </Button>
+    </View>
+    <View style={styles.gap} />
+    <View style={styles.column}>
+        <Button>
+            Button in a column
+        </Button>
+    </View>
+</View>
+```
+
+Layouts often specify a specific width of button.  When implementing such
+designs use `minWidth` instead of `width`.  It prevents a button from
+resizing to fit content.  This is important for international sites since
+sometimes strings for UI elements can be much longer than their English
+counterparts.
```

Comment:
I'm confused by this part. Is "It prevents a button..." referring to `minWidth` or `width`? Is `badButton` supposed to be with `minWidth`?